src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -456,52 +456,28 @@ private CommandInfo GetNextFromPath()
                     "Trying to resolve the path as an PSPath");
 
                 // Find the match if it is.
-
-                Collection<string> resolvedPaths = new Collection<string>();
-
-                try
-                {
-                    Provider.CmdletProvider providerInstance;
-                    ProviderInfo provider;
-                    resolvedPaths =
-                        _context.LocationGlobber.GetGlobbedProviderPathsFromMonadPath(_commandName, false, out provider, out providerInstance);
-                }
-                catch (ItemNotFoundException)
+                // Try literal path resolution if it is set to run first
+                if (_commandResolutionOptions.HasFlag(SearchResolutionOptions.ResolveLiteralThenPathPatterns))
                 {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "The path could not be found: {0}",
-                        _commandName);
-                }
-                catch (DriveNotFoundException)
-                {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "A drive could not be found for the path: {0}",
-                        _commandName);
-                }
-                catch (ProviderNotFoundException)
-                {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "A provider could not be found for the path: {0}",
-                        _commandName);
-                }
-                catch (InvalidOperationException)
-                {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "The path specified a home directory, but the provider home directory was not set. {0}",
-                        _commandName);
+                    var path = GetNextLiteralPathThatExists(_commandName, out _);
+
+                    // This can return null, but that is expected.
+                    // The searcher will continue, if it can.
+                    return GetInfoFromPath(path);
                 }
-                catch (ProviderInvocationException providerException)
+
+                Collection<string> resolvedPaths = new Collection<string>();
+                if (WildcardPattern.ContainsWildcardCharacters(_commandName))
                 {
-                    CommandDiscovery.discoveryTracer.TraceError(
-                        "The provider associated with the path '{0}' encountered an error: {1}",
-                        _commandName,
-                        providerException.Message);
+                    resolvedPaths = GetNextFromPathUsingWildcards(_commandName, out _);
                 }
-                catch (PSNotSupportedException)
+
+                // Try literal path resolution if wildcards are enable first and wildcard search failed
+                if (!_commandResolutionOptions.HasFlag(SearchResolutionOptions.ResolveLiteralThenPathPatterns) &&
```

Comment:
This flag check seems unnecessary.  If the flag is set, it would have returned by line 466, after that `if` statement, it should never have this flag set.

Reply:
Both searches can return nothing, removing this check will hurt perf, but not functionally needed.

Reply:
Updated previous code to not return if it found nothing.